qcodes_contrib_drivers/drivers/CopperMountain/M5180.py

```diff
@@ -0,0 +1,564 @@
+                "{}".format(val, stop)
+            )
+        self.update_lin_traces()
+
+    def _set_span(self, val: float) -> None:
+        """Sets frequency span and updates linear trace parameters.
+
+        Args:
+            val (float): frequency span to be set
+        """
+        self.write("SENS1:FREQ:SPAN {}".format(val))
+        self.update_lin_traces()
+
+    def _set_center(self, val: float) -> None:
+        """Sets center frequency and updates linear trace parameters.
+
+        Args:
+            val (float): center frequency to be set
+        """
+        self.write("SENS1:FREQ:CENT {}".format(val))
+        self.update_lin_traces()
+
+    def _set_npts(self, val: int) -> None:
+        """Sets number of points and updates linear trace parameters.
+
+        Args:
+            val (int): number of points to be set
+        """
+        self.write("SENS1:SWE:POIN {}".format(val))
+        self.update_lin_traces()
+
+    def _get_trigger(self) -> str:
+
+        r = self.ask('TRIG:SOUR?')
+
+        if r.lower()=='int':
+            return 'internal'
+        elif r.lower()=='ext':
+            return 'external'
+        elif r.lower()=='man':
+            return 'manual'
+        else:
+            return 'bus'
+
+    def _set_trigger(self, trigger: str) -> None:
+
+        self.write('TRIG:SOUR '+trigger.upper())
+
+    def get_s(self) -> Tuple[np.ndarray]:
+        """
+        Return S parameter as magnitude in dB and phase in rad.
+
+        Returns:
+            Tuple[np.ndarray]: frequency [GHz],
+                               s11 magnitude [dB], s11 phase [rad],
+                               s12 magnitude [dB], s12 phase [rad],
+                               s21 magnitude [dB], s21 phase [rad],
+                               s22 magnitude [dB], s22 phase [rad]
+        """
+
+        self.write('CALC1:PAR:COUN 4') # 4 trace
+        self.write('CALC1:PAR1:DEF S11') # Choose S11 for trace 1
+        self.write('CALC1:PAR2:DEF S12') # Choose S12 for trace 2
+        self.write('CALC1:PAR3:DEF S21') # Choose S21 for trace 3
+        self.write('CALC1:PAR4:DEF S22') # Choose S22 for trace 4
+        self.write('CALC1:TRAC1:FORM SMITH')  # Trace format
+        self.write('CALC1:TRAC2:FORM SMITH')  # Trace format
+        self.write('CALC1:TRAC3:FORM SMITH')  # Trace format
+        self.write('CALC1:TRAC4:FORM SMITH')  # Trace format
+        self.write('TRIG:SEQ:SING') # Trigger a single sweep
+        self.ask('*OPC?') # Wait for measurement to complete
+
+        # Get data as string
+        freq = self.ask("SENS1:FREQ:DATA?")
+        s11 = self.ask("CALC1:TRAC1:DATA:FDAT?")
+        s12 = self.ask("CALC1:TRAC2:DATA:FDAT?")
+        s21 = self.ask("CALC1:TRAC3:DATA:FDAT?")
+        s22 = self.ask("CALC1:TRAC4:DATA:FDAT?")
+
+        # Get data as numpy array
+        freq = np.fromstring(freq, dtype=float, sep=',')
+        s11 = np.fromstring(s11, dtype=float, sep=',')
+        s11 = s11[0::2] + 1j*s11[1::2]
```

Comment:
Why do I get the error `qcodes_contrib_drivers/drivers/CopperMountain/M5180.py:518:30: error: Unsupported operand types for * ("complex" and "str")`? Why is `s11` of type `str`?

Reply:
This I figured out, I do not change the type of a variable any longer.